### PR TITLE
Update DBT-2 packaging source

### PR DESCRIPTION
**Pull request by hannahms**
Updates the rpm install of the necesary DBT-2 packages retrieved from the PGDG repo, when applicable. The test scripts and removal tasks have also been updated to reflect this new packaging. 

- The previous `dbt2-scripts`, `dbt2-db`, `dbt2-pgsql-plpgsql`, `dbt2-client`, `dbt2-driver`, and `dbt2-exec` packages are replaced by the PGDG `dbt2-common` package. 
- The `dbttools` package is replaced by PGDG `dbttools` package. 
- The `dbt2-pgsql-c_{pg_version}` package is still required for PG v11 and v12 (v15 is not included in the `dbt2-packaging` repo nor the PGDG packaging), but for PG v13 and v14, the PGDG package `dbt2-pg{pg_version}-extensions` can replace the other packaging.

Validation for the installed items for each package can be found under the 'Files' header: [`dbt2-common`](https://rhel.pkgs.org/8/postgresql-common-aarch64/dbt2-common-0.49.1-1.rhel8.aarch64.rpm.html), [`dbttools`](https://rhel.pkgs.org/8/postgresql-common-x86_64/dbttools-0.3.2-1.rhel8.noarch.rpm.html) and [`dbt2-pg14-extensions`](https://rhel.pkgs.org/8/postgresql-14-x86_64/dbt2-pg14-extensions-0.50.1-1.rhel8.x86_64.rpm.html).

In the `setup_dbt2_driver` role, the old method of retrieving the `dbt2-packaging` package with the DBT-2 version in the URL stayed present for v11 and v12 installation. The more recent `dbt2-packaging` releases (beyond current edb-ansible default 0.48.7) only include the AppImage. Aside from this package, this role will be compatible with ARM architecture.

### CODE REVIEW

**Comment by mw2q** [comment]
This looks to reasonable to to me.  I think we should test it before merging.  I have some vague ideas on how, but need more thinking time to think through it...

**Comment by hannahms** [comment]
Were you able to run any tests locally?

**Comment by mw2q** [comment]
I've tried to run this by hand with edb-benchmarks/aws-dbt2-biganimal and the prepare step is failing.  Something's changed in edb-terraform.  will investigate next chance i get, unless @bryan-bar knows offhand.

```
TASK [Populate DBT2 database] ********************************************************************************************************
fatal: [dbt2-driver]: FAILED! => {"msg": "The field 'environment' has an invalid value, which includes an undefined variable. The error was: 'dict object' has no attribute 'public_ip'. 'dict object' has no attribute 'public_ip'\n\nThe error appears to be in '/home/mark.wong/src/edb-benchmarks/aws-dbt2-biganimal/prepare/playbook-dbt2-build-db.yml': line 16, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Populate DBT2 database\n      ^ here\n"}
```

**Comment by mw2q** [comment]
I hot patched some edb-benchmark execution changes, and I think the pl/pgsql stored functions are missing.  next on my list to investigate.

**Comment by hannahms** [comment]
Those are available under the command `/usr/bin/dbt2-pgsql-load-stored-procs`

**Comment by mw2q** [comment]
Right, that's the script to load them, but I think the stored procedures
themselves might be missing.  (Still haven't looked myself yet.)

**Comment by mw2q** [comment]
The `dbt2-pg*-extensions` packages has the pl/pgsql and C stored procedures in them.  That is only awkward for the pl/pgsql procedures because those are not tied to postgres versions.  Next step is to make time to see if installing any of them let me run one of our canned benchmarks.

**Comment by hannahms** [comment]
I have update the packaging to include the `install_dbserver` role for the `dbt2_driver` group to install the PG packages on the driver to access `pg_config`.

**Comment by mw2q** [comment]
Thanks.  I'm not sure when I can get back to it, but if someone else is making time, I was using the aws-dbt2-biganimal test in edb-benchmarks to proof it by hand.

**Review by mw2q — CHANGES_REQUESTED** [review]
I've made another pass.  Aside from the previous comments, I think this is works.

We need to be careful about merging/releasing these changes because they require changes in edb-benchmarks playbooks that are using dbt-2:

- updating deploy playbooks to `setup_repo` and `install_dbserver`
- removal of "-u" and flags for build-db and run-workload scripts
- additional `environment:` ansible setting to set `PATH: "/usr/pgsql-{{ pg_version }}/bin:{{ ansible_env.PATH }}"`

File: roles/setup_dbt2/tasks/dbt2_install_packages_db.yml [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
+  when: ansible_os_family == "RedHat"
+  become: true
+
+- name: Install DBT-2 pg extensions package for DBT-2 Database for PG v13 or v14
+  ansible.builtin.package:
+    name: "dbt2-pg{{ pg_version }}-extensions"
+    state: present
+  when:
+    - ansible_os_family == "RedHat"
+    - pg_version|string in ['13', '14']
```
[the remark this answers is not in the document]
Comment (hannahms) [reply roles/setup_dbt2/tasks/dbt2_install_packages_db.yml:15]: We can do that by adding a supported_versions in the `/defaults/main.yml` and checking those before running any script further, but in this context, the `dbt2-pg13-extensions`/`dbt2-pg14-extensions` repo packages are only supporting v13 and v14, so this version check is to determine whether to download from the github repo if it is not v13 or v14, or to download the package with the correct version in the package name. I think the role will still work as expected for v12, v15, etc.

**Comment by mw2q** [comment]
Right, while the edb-ansible role could work to install from either repo
location, individual tests in edb-benchmark will have to be written
differently depending on which version of dbt2 is installed.  I think it'll
be less work for us if we stick to the pgdg repository packages.

**Comment by hannahms** [comment]
I'm circling back to this and I think I have some confusion on how to move forward when the pgdg repo only contains packages for v13 and v14. The version checks are present for that reason, and if I remove them then the roles will only work for v13 and v14, which to me is not ideal. I can add the environment changes, but that doesn't solve the version issues, especially if the benchmarks need them to be installed consistently.

**Comment by mw2q** [comment]
I think it's time to move to the AppImages:
https://github.com/osdldbt/dbt2/releases

I'm hoping v0.53.2 is ready to go.

The only advantage of the packages at this stage is the availability of C
stored functions.  Our immediate needs don't need them, so we can revisit
how to use C stored functions at a later time.

So the setup playbooks should be very similar to the touchstone-tools
playbooks.

**Comment by hannahms** [comment]
AppImage has been added as source of DBT-2, it is placed into `dbt2_path: "/usr/bin/dbt2"`. All commands can be run with `dbt2 *command*`.

File: roles/setup_dbt2/README.md [hunk 2]
```diff
@@ -29,9 +30,15 @@ When executing the role via ansible these are the required variables:
   link for available versions:
   https://github.com/osdldbt/dbttools-packaging/releases
 
+  * ***dbt2_path***
+
+  This is the location where the DBT-2 AppImage will be installed.
+  For ease of use, try to set this location within a directory along the executable `PATH`.
+  Unless necessary, there shouldn't be any reason to change this default.
+
   * ***pg_version***
```
Comment (mw2q) [inline roles/setup_dbt2/README.md:39]: pg_version parameter is no longer needed by dbt2

File: roles/setup_dbt2/README.md [hunk 3]
```diff
@@ -16,9 +17,9 @@ When executing the role via ansible these are the required variables:
 
   * ***dbt2_version***
 
-  These playbooks can install any version of DBT-2 that is packaged from GitHub.
+  These playbooks can install any version of DBT-2 that is packaged into an AppImage from GitHub.
   See the following link for available versions:
-  https://github.com/osdldbt/dbt2-packaging/releases
+  https://github.com/osdldbt/dbt2/releases
 
   * ***dbttools_version***
```
Comment (mw2q) [inline roles/setup_dbt2/README.md:24]: dbttools_version parameters is no longer needed, it's in the dbt2 appimage.

File: plugins/lookup/supported_roles.py [hunk 4]
```diff
@@ -77,11 +77,9 @@
     ],
     'dbt2_driver': [
         'setup_repo',
```
Comment (mw2q) [inline plugins/lookup/supported_roles.py]: i don't think setup_repo is needed for dbt2.  it just needs fuse, and sysstat, and perf.  That's all in the distro's default repos, right?

File: roles/setup_dbt2/defaults/main.yml [hunk 5]
```diff
@@ -1,11 +1,16 @@
 ---
-dbt2_version: "0.48.7"
+dbt2_version: "0.53.2"
 dbttools_version: "0.3.1"
```
Comment (mw2q) [inline roles/setup_dbt2/defaults/main.yml:3]: ditto about dbttools_version here

File: roles/setup_dbt2/defaults/main.yml [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
 driver_pub_key: ""
 pg_dbt2_dbname: dbt2
 pg_version: 14
```
Comment (mw2q) [inline roles/setup_dbt2/defaults/main.yml:6]: ditto about pg_version here

File: roles/setup_dbt2/tasks/dbt2_install_packages_client.yml [hunk 7]
```diff
@@ -1,17 +1,20 @@
...
-  ansible.builtin.package:
-    name: dbt2-common
-    state: present
-  when: ansible_os_family == "RedHat"
-  become: true
-
-- name: Install additional supporting packages
+- name: Install DBT-2 Client supporting packages
   ansible.builtin.package:
     name:
```
Comment (mw2q) [inline roles/setup_dbt2/tasks/dbt2_install_packages_client.yml:4]: probably should add sysstat here.  i don't know why we didn't think of that before.

File: roles/setup_dbt2/tasks/dbt2_install_packages_db.yml [hunk 8]
```diff
@@ -1,37 +1,20 @@
...
-    disable_gpg_check: true
-  when:
-    - ansible_os_family == "RedHat"
-    - pg_version|string not in ['13', '14']
-  become: true
-
-- name: Install additional supporting packages
+- name: Install DBT-2 supporting packages
   ansible.builtin.package:
     name:
```
Comment (mw2q) [inline roles/setup_dbt2/tasks/dbt2_install_packages_db.yml:4]: ditto about adding sysstat here

File: roles/setup_dbt2/tasks/dbt2_install_packages_driver.yml [hunk 9]
```diff
@@ -42,14 +13,25 @@
     mode: '0755'
   become: true
 
-- name: Install additional supporting packages
+- name: Install DBT-2 Driver supporting packages
   ansible.builtin.package:
     name:
```
Comment (mw2q) [inline roles/setup_dbt2/tasks/dbt2_install_packages_driver.yml:18]: don't need python3-docutils anymore, it's in the appimage.  also don't need postgresql anymore, that's also in the appimage.  also don't need dbttools, that's in the appimage.

File: roles/setup_dbt2/tasks/rm_dbt2_install_config.yml [hunk 10]
```diff
@@ -5,31 +5,10 @@
...
-  ansible.builtin.package:
-    name: dbt2-common
+- name: Remove DBT-2 AppImage
+  ansible.builtin.file:
+    path: "{{ dbt2_path }}"
     state: absent
-  when: ansible_os_family == "RedHat"
   become: true
 
 - name: Remove additional supporting packages
```
Comment (mw2q) [inline roles/setup_dbt2/tasks/rm_dbt2_install_config.yml:14]: let's add sysstat here too

File: roles/setup_dbt2_client/README.md [hunk 11]
```diff
@@ -26,13 +24,19 @@ When executing the role via ansible these are the required variables:
...
-  https://github.com/osdldbt/dbt2-packaging/releases
+  https://github.com/osdldbt/dbt2/releases
+
+  * ***dbt2_path***
+
+  This is the location where the DBT-2 AppImage will be installed.
+  For ease of use, try to set this location within a directory along the executable `PATH`.
+  Unless necessary, there shouldn't be any reason to change this default.
 
   * ***pg_version***
```
Comment (mw2q) [inline roles/setup_dbt2_client/README.md:37]: ditto, don't need pg_version here

File: roles/setup_dbt2_client/tasks/rm_dbt2_client_install_config.yml [hunk 12]
```diff
@@ -14,11 +14,10 @@
...
-  ansible.builtin.dnf:
-    name: 'dbt2-common'
+- name: Remove DBT-2 AppImage
+  ansible.builtin.file:
+    path: "{{ dbt2_path }}"
     state: absent
-  when: ansible_os_family == "RedHat"
   become: true
 
 - name: Remove additional supporting packages
```
Comment (mw2q) [inline roles/setup_dbt2_client/tasks/rm_dbt2_client_install_config.yml:23]: let's add sysstat here

File: roles/setup_dbt2_driver/README.md [hunk 13]
```diff
@@ -19,19 +19,25 @@ When executing the role via ansible these are the required variables:
 
   * ***dbt2_version***
 
-  These playbooks can install any version of DBT-2 that is packaged from GitHub.
+  These playbooks can install any version of DBT-2 that is packaged into an AppImage from GitHub.
   See the following link for available versions:
-  https://github.com/osdldbt/dbt2-packaging/releases
+  https://github.com/osdldbt/dbt2/releases
 
   * ***dbttools_version***
```
Comment (mw2q) [inline roles/setup_dbt2_driver/README.md:26]: ditto, don't need dbttools_version here

File: roles/setup_dbt2_driver/defaults/main.yml [hunk 14]
```diff
@@ -6,9 +6,12 @@ supported_os:
   - Rocky8
   - AlmaLinux8
 
-dbt2_version: "0.48.7"
+dbt2_version: "0.53.2"
 dbttools_version: "0.3.1"
```
Comment (mw2q) [inline roles/setup_dbt2_driver/defaults/main.yml:10]: ditto don't need dbttools_version here

File: roles/setup_dbt2_driver/tasks/rm_dbt2_driver_install_config.yml [hunk 15]
```diff
@@ -15,35 +15,7 @@
...
-    name:
-      - "dbt2-pg{{ pg_version }}-extensions"
-    state: absent
-  when:
-    - ansible_os_family == "RedHat"
-    - pg_version|string not in ['13', '14']
+    - "{{ dbt2_path }}"
   become: true
 
 - name: Remove additional supporting packages
```
Comment (mw2q) [inline roles/setup_dbt2_driver/tasks/rm_dbt2_driver_install_config.yml:21]: don't need python3-docutils or dbttools here.

File: tests/tests/test_setup_dbt2.py [hunk 16]
```diff
@@ -1,33 +1,36 @@
...
-    get_pg_version,
     get_primary,
     get_pg_unix_socket_dir
 )
 
 
 def test_setup_dbt2_packages():
     host = get_primary()
-    pg_version = get_pg_version()
     packages = [
```
Comment (mw2q) [inline tests/tests/test_setup_dbt2.py:12]: assuming you want sysstat here too

File: tests/tests/test_setup_dbt2_client.py [hunk 17]
```diff
@@ -4,21 +4,32 @@
...
     get_dbt2_client,
-    get_pg_version,
-    get_primary,
-    get_pg_unix_socket_dir
 )
 
+
 def test_setup_dbt2_client_packages():
     host = get_dbt2_client()[0]
     packages = [
```
Comment (mw2q) [inline tests/tests/test_setup_dbt2_client.py:12]: assuming you want sysstat here

File: tests/tests/test_setup_dbt2_driver.py [hunk 18]
```diff
@@ -1,28 +1,37 @@
...
     get_dbt2_driver,
-    get_pg_version,
-    get_primary,
-    get_pg_unix_socket_dir
 )
 
+
 def test_setup_dbt2_driver_packages():
     host = get_dbt2_driver()[0]
     packages = [
```
Comment (mw2q) [inline tests/tests/test_setup_dbt2_driver.py:11]: assuming we want sysstat here.  don't need dbttools here anymore.

**Review by mw2q — CHANGES_REQUESTED** [review]

File: plugins/lookup/supported_roles.py (hunk 4 above)
Comment (hannahms) [reply plugins/lookup/supported_roles.py]: correct, removed

**Comment by hannahms** [comment]
Changes made to add `sysstat` package, remove `setup_repo` requirement for `dbt2_driver`, remove `dbttools` information and remove `pg_version`.

**Review by mw2q — APPROVED** [review]
this looks good to me

**Comment by hannahms** [comment]
Should we wait to merge until it's been tested against a benchmark? Some of the playbooks will need to be updated from `dbt2-pgsql-build-db` to `dbt2 pgsql-build-db`, for example, and I want to make sure merging doesn't mess anything up

**Pull request merged**